Description: Fix endianness regression from 2.7 to 2.8
Author: HW42 <hw42@ipsumj.de>
Bug: https://github.com/mm2/Little-CMS/issues/110
Bug-Debian: https://bugs.debian.org/847595
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/cmsnamed.c
+++ b/src/cmsnamed.c
@@ -192,18 +192,10 @@
 static
 void strFrom16(char str[3], cmsUInt16Number n)
 {
-    // Assiming this would be aligned
-    union {
-
-       cmsUInt16Number n;
-       char str[2];
-       
-    } c;
-
-    c.n = n;  // Always big endian in this case
-
-    str[0] = c.str[0]; str[1] = c.str[1]; str[2] = 0;
-
+    // n is always big endian, see strTo16.
+    str[0] = (n >> 8) & 0xff;
+    str[1] = n & 0xff;
+    str[2] = 0;
 }
 
 // Add an ASCII entry. Do not add any \0 termination (ICC1v43_2010-12.pdf page 61)
